electron/preload/index.ts

```diff
@@ -203,9 +202,15 @@ contextBridge.exposeInMainWorld("electronStore", {
 
 contextBridge.exposeInMainWorld("ipcRenderer", {
   on: ipcRenderer.on,
-  removeListener: ipcRenderer.removeListener,
   receive: (channel: string, callback: (...args: unknown[]) => void) => {
-    ipcRenderer.on(channel, (event, ...args) => callback(...args));
+    // this creates a constant copy of the callback, thus ensuring that the removed listener is the same as the one added
+    const subscription = (event: IpcRendererEvent, ...args: unknown[]) =>
+      callback(...args);
+
+    ipcRenderer.on(channel, subscription);
+    return () => {
+      ipcRenderer.removeListener(channel, subscription);
+    };
```

Comment:
Essentially uses this approach here - https://github.com/reZach/secure-electron-template/issues/43#issuecomment-771719565